src/common/cli_wrapper.h

```diff
@@ -0,0 +1,275 @@
+   */
+  void switchGroup(const std::string &name = "");
+
+  // Parse command-line arguments. Handles --help and --version options
+  void parse(int argc, char **argv);
+
+  // Check if an option has been defined (not necessarily parsed)
+  bool has(const std::string &key) const;
+
+  // Get the current value for the option
+  template <typename T>
+  T get(const std::string &key) const {
+    ABORT_IF(
+        vars_.count(key) == 0, "An option with key '{}' is not defined", key);
+    return vars_.at(key)->as<T>();
+  }
+
+  YAML::Node getConfig() const;
+
+  void setConfig(const YAML::Node &config);
+
+  /**
+   * @brief Overwrite values for unparsed options
+   *
+   * Default values are overwritten with the options found in the config
+   * provided as the argument, while parsed command-line options remain
+   * unchanged
+   *
+   * @param node YAML config with new default values for options
+   */
+  void overwriteDefault(const YAML::Node &node);
+
+private:
+  template <
+      typename T,
+      CLI::enable_if_t<!CLI::is_bool<T>::value && !CLI::is_vector<T>::value,
+                       CLI::detail::enabler> = CLI::detail::dummy>
+  CLI::Option *add_option(const std::string &key,
+                          const std::string &args,
+                          const std::string &help,
+                          T val,
+                          bool defaulted,
+                          bool addToConfig) {
+    // std::cerr << "CLI::add(" << key << ") " << std::endl;
+
+    if(addToConfig)
+      config_[key] = val;
+    vars_.insert(std::make_pair(key, std::make_shared<any_type>(val)));
+
+    CLI::callback_t fun = [this, key](CLI::results_t res) {
+      // std::cerr << "CLI::callback(" << key << ") " << std::endl;
+      auto &var = vars_[key]->as<T>();
+      auto ret = CLI::detail::lexical_cast(res[0], var);
+      config_[key] = var;
+      return ret;
+    };
+
+    auto opt = app_->add_option(args, fun, help, defaulted);
+    opt->type_name(CLI::detail::type_name<T>());
+    if(!currentGroup_.empty())
+      opt->group(currentGroup_);
+    if(defaulted) {
+      std::stringstream ss;
+      ss << val;
+      opt->default_str(ss.str());
+    }
+
+    opts_.insert(std::make_pair(key, opt));
+    return opts_[key];
+  }
+
+  template <typename T,
+            CLI::enable_if_t<CLI::is_vector<T>::value,
+                             CLI::detail::enabler> = CLI::detail::dummy>
+  CLI::Option *add_option(const std::string &key,
+                          const std::string &args,
+                          const std::string &help,
+                          T val,
+                          bool defaulted,
+                          bool addToConfig) {
+    // std::cerr << "CLI::add(" << key << ") as vector" << std::endl;
+
+    if(addToConfig)
+      config_[key] = val;
+    vars_.insert(std::make_pair(key, std::make_shared<any_type>(val)));
+
+    CLI::callback_t fun = [this, key](CLI::results_t res) {
```

Comment:
One-line comment what this function does and what it is for

Reply:
Done